### PR TITLE
Return items of multi-select in order of selection

### LINKED ISSUE #464: Multi-select should output items in order in which they were selected
In multi-select, items are currently returned according to some internal order, rather than in the order in which they were selected.

For applications where the item order matters, it would be useful to return the items in order of selection. If algorithmic sorting of any kind is desired, it is always possible to do so in a post-processing step. But it is not possible, on the other hand, to recover the order of manual selections, unless skim provides this information.

My use-case involves selecting multiple files for processing where order matters and cannot be inferred from the file names or their contents.

Thanks!

### PR DESCRIPTION
**Pull request by otto-dev**
Fixes #464. Also preserves the order of `--pre-select-items` when returning them unchanged. The solution is very simple and natural. I have relied on this behavior a lot during the last few days, and it works fine.

The introduced `linked-hash-map` dependency will continue to receive bug-fixes and updates to rust versions, but there are not plans to add further features to the library.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -35,10 +35,11 @@ rayon = "1.5.3"
...
 bitflags = "1.3.2"
 timer = "0.2.0"
-chrono = "0.4.22"
-crossbeam = "0.8.2"
-beef = "0.5.2" # compact cow
-defer-drop = "1.3.0"
+chrono = "0.4"
+crossbeam = "0.8.0"
+beef = "0.5.0" # compact cow
+defer-drop = "1.0.1"
```
Comment (yazgoo) [inline Cargo.toml]: why downgrading some packages

**Review by yazgoo — APPROVED** [review]

**Review by yazgoo — CHANGES_REQUESTED** [review]
see previous question

File: Cargo.toml (hunk 1 above)
Comment (LoricAndre) [reply Cargo.toml]: should be fixed

**Review by yazgoo — APPROVED** [review]
looks ok to me

**Pull request merged**